source/common/http/conn_manager_impl.cc

```diff
@@ -836,6 +847,18 @@ ConnectionManagerImpl::ActiveStream::requestHeadersForTags() const {
   return connection_manager_.config_.tracingConfig()->request_headers_for_tags_;
 }
 
+void ConnectionManagerImpl::ActiveStream::callHighWatermarkCallbacks() {
+  for (DownstreamWatermarkCallbacks* callbacks : watermark_callbacks_) {
+    callbacks->onAboveWriteBufferHighWatermark();
```

Comment:
Yes if upstream connection dies, the reset chain will start from that side and be cleaned up properly.